src/Components/Components/test/ComponentBaseTest.cs

```diff
@@ -372,6 +372,37 @@ public async Task DoesNotRenderAfterOnInitAsyncTaskIsCancelledUsingCancellationT
         Assert.NotEmpty(renderer.Batches);
     }
 
+    [Fact]
+    public async Task ErrorBoundaryHandlesOnInitializedAsyncReturnFaultedTask()
+    {
+        // Arrange
+        var renderer = new TestRenderer();
+        TestErrorBoundary capturedBoundary = null;
+
+        // Create root component that wraps the TestComponentErrorBuildRenderTree in an TestErrorBoundary
+        var rootComponent = new TestComponent();
+        rootComponent.ChildContent = builder =>
+        {
+            builder.OpenComponent<TestErrorBoundary>(0);
+            builder.AddComponentParameter(1, nameof(TestErrorBoundary.ChildContent), (RenderFragment)(childBuilder =>
+            {
+                childBuilder.OpenComponent<TestComponentErrorBuildRenderTree>(0);
+                childBuilder.CloseComponent();
+            }));
+            builder.AddComponentReferenceCapture(2, inst => capturedBoundary = (TestErrorBoundary)inst);
+            builder.CloseComponent();
+        };
+
+        // Act
+        var rootComponentId = renderer.AssignRootComponentId(rootComponent);
+        await renderer.RenderRootComponentAsync(rootComponentId);
+
+        // Assert
+        Assert.NotNull(capturedBoundary);
+        Assert.NotNull(capturedBoundary!.ReceivedException);
+        Assert.Equal(typeof(InvalidTimeZoneException), capturedBoundary!.ReceivedException.GetType());
+    }
+
```

Comment:
Can we add a test for `ComponentBase` to capture the new behavior without ErrorBoundary? (That StateHasChanged doesn't get invoked with a faulted task)

Are there other similar cases that we need to look for? (Like CallOnParametersSetAsync) https://github.com/dotnet/aspnetcore/blob/main/src/Components/Components/src/ComponentBase.cs#L320-L322